pytorch_lightning/callbacks/progress.py

```diff
@@ -293,7 +293,15 @@ def init_test_tqdm(self) -> tqdm:
     def on_sanity_check_start(self, trainer, pl_module):
         super().on_sanity_check_start(trainer, pl_module)
         self.val_progress_bar = self.init_sanity_tqdm()
-        self.val_progress_bar.total = convert_inf(trainer.num_sanity_val_steps * len(trainer.val_dataloaders))
+
+        # There will be circular imports if we move the following import to the
+        # top of the file.
+        from pytorch_lightning.trainer.data_loading import has_len
```

Comment:
then move the function `has_len` to a module where it ll be independent, e.g. pl/utilities/data.py

Reply:
Thank you. I've fixed it.